src/test/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerRunFingerprintFacetTest.java

```diff
@@ -46,14 +46,17 @@ public void test_readResolve() throws Exception {
         FreeStyleProject p = rule.createFreeStyleProject("test");
         FreeStyleBuild b = rule.assertBuildStatusSuccess(p.scheduleBuild2(0));
         
-        ContainerRecord r = new ContainerRecord("192.168.1.10", "cid", null, "magic", System.currentTimeMillis(), Collections.<String, String>emptyMap());
-        DockerFingerprints.addRunFacet(IMAGE_ID, r, b);
-        Fingerprint fingerprint = DockerFingerprints.of(IMAGE_ID);
-        DockerRunFingerprintFacet facet = (DockerRunFingerprintFacet) fingerprint.getFacets().iterator().next();
+        ContainerRecord r1 = new ContainerRecord("192.168.1.10", "cid", IMAGE_ID, "magic", System.currentTimeMillis(), Collections.<String, String>emptyMap());
+        DockerFingerprints.addRunFacet(r1, b);
 
-        Assert.assertNull(r.getImageId());
+        Fingerprint fingerprint = DockerFingerprints.of(IMAGE_ID);
+        DockerRunFingerprintFacet facet = new DockerRunFingerprintFacet(fingerprint, System.currentTimeMillis(), IMAGE_ID);
+        ContainerRecord r2 = new ContainerRecord("192.168.1.10", "cid", null, "magic", System.currentTimeMillis(), Collections.<String, String>emptyMap());
+        facet.add(r2);
+        
+        Assert.assertNull(r2.getImageId());
         facet.readResolve();
```

Comment:
That would be better, but I don't know what I need to do to mimic that. I'm sure I'd figure it out, but I suppose there's a bit of a tradeoff in terms of the amount of time I have to spend writing a test Vs the level of potential risk.

Personally, I wouldn't have opened the `transient` + `readResolve` can of worms at all.